models/posts.js

```diff
@@ -150,58 +141,52 @@ module.exports = (sequelize, DataTypes) => {
       } ).then(function(result){
         callback(null,result);
       }).catch(function(error){
-
-        return callback({
-          message:error.message
-        });
+        return callback(error,null);
       });
     };
 
-    blog.enterPost=function(req,callback)
+    post.enterPost=function(req,callback)
     {
-      blog.create(
+      post.create(
         {
           id:req.body.id,
           title:req.body.title,
-          category_id:req.body.category_id,
-          User_id:req.body.User_id,
+          categoryId:req.body.categoryId,
+          userId:req.body.userId,
           content:req.body.content,
         }).then(function(enter)
         {
           callback(null,enter);
         })  .catch(function(error){
-
-          return callback({
-            message:error.message
-          });
+          return callback(error,null);
         });
       };
 
-      blog.getAllblog=function(model_ref,req,callback)
+      post.getAllPosts=function(model_ref,req,callback)
```

Comment:
Make functionality to handle single post here itself.
When you create separate function ie. getSinglePost and getAllPost the querying remains. You just need to add where condition. 
also handle pagination's limit and offset

Same applies to other models getAll functions